rasa/nlu/regex_message_handler.py

```diff
@@ -0,0 +1,292 @@
+                f"but found {unknown_entity_types}. "
+                f"Continuing without unknown entity types. ",
+                docs=DOCS_URL_STORIES,
+            )
+            parsed_entities = {
+                key: value
+                for key, value in parsed_entities
+                if key not in unknown_entity_types
+            }
+
+        # convert them into the list of dictionaries that we expect
+        entities: List[Dict[Text, Any]] = []
+        for entity_type, entity_values in parsed_entities.items():
+            if not isinstance(entity_values, list):
+                entity_values = [entity_values]
+            for entity_value in entity_values:
+                entities.append(
+                    {
+                        ENTITY_ATTRIBUTE_TYPE: entity_type,
+                        # TODO: in the original implementation these are start and
+                        # end indices of the *substring that matches the entity regex*
+                        # ... but it should be indices of tokens to which these
+                        # entities belong. There are no such tokens. Hence, I think we
+                        # should just skip those indices. But can we, or will something
+                        # complain? If something expects indices (despite not being
+                        # able to use them because there's no text), then we could
+                        # instead use more explicit "unknown" indicators (e.g. -1)
+                        # ENTITY_ATTRIBUTE_START: sidx,
+                        # ENTITY_ATTRIBUTE_END: eidx,  # can't be more specific
+                        ENTITY_ATTRIBUTE_VALUE: entity_value,
+                    }
+                )
+        return entities
+
+    @staticmethod
+    def _parse_optional_confidences(match: Match) -> float:
+        """Extracts the optional confidence information from the given pattern match.
+
+        If no confidence is specified, then this method returns the maximum
+        confidence `1.0`.
+        If a confidence is specified but extraction fails, then this method defaults
+        to a confidence of `0.0`.
+
+        Args:
+          match: a match produced by `self.pattern`
+          domain: the domain
+        Returns:
+          some confidence value
+        """
+        confidence_str = match.group(PREDICTED_CONFIDENCE_KEY)
+        if confidence_str is None:
+            return 1.0
+        try:
+            confidence_str = confidence_str.strip()[1:]  # remove the "@"
+            try:
+                confidence = float(confidence_str)
+            except ValueError:
+                confidence = 0.0
+                raise ValueError(
+                    f"Expected confidence to be a non-negative decimal number but "
+                    f"found {confidence}. Continuing with 0.0 instead."
+                )
+            # NOTE: This is new. Before, we would've just passed on this value,
+            # whatever it was.
```

Comment:
Curious why it's needed.

Reply:
The default is 1.0 which somehow implies that this should be bounded by 1.

Reply:
Daksh just confirmed this is reasonable :)